relay.go

```diff
@@ -188,17 +276,23 @@ func (r *Relayer) addRelayItem(isOriginator bool, id, remapID uint32, destinatio
 	if isOriginator {
 		items = r.outbound
 	}
-
-	r.Lock()
-	items[id] = item
-	r.Unlock()
+	item.Timer = time.AfterFunc(ttl, func() { r.timeoutRelayItem(items, id, isOriginator) })
+	items.Add(id, item)
 	return item
 }
 
-func (r *Relayer) removeRelayItem(items map[uint32]relayItem, id uint32) {
-	r.Lock()
-	delete(items, id)
-	r.Unlock()
+func (r *Relayer) timeoutRelayItem(items *relayItems, id uint32, isOriginator bool) {
+	items.Entomb(id, _relayTombTTL)
+	if isOriginator {
+		// TODO: As above. What's the span in the error frame for?
+		r.conn.SendSystemError(id, nil, ErrTimeout)
+	}
+	r.decPending()
```

Comment:
(feel free to add TODO for this)
we should be more defensive when decrementing this count -- we should only do it if `Entomb` set tomb=true for the first time. Same thing below, we should only decrement if delete actually removed the item and it wasn't already pending. Otherwise, we might want to log a warning since it would mean we have a bug.